lib/plugins/aws/package/lib/mergeIamTemplates.js

```diff
@@ -39,8 +39,12 @@ module.exports = {
 
     const iamRole = _.get(this.serverless.service.provider.iam, 'role', {});
 
-    // resolve early if provider level role is provided
-    if (typeof iamRole === 'string') {
+    // resolve early if provider level role is provided as string or CF intrinsic function
+    if (
+      typeof iamRole === 'string' ||
+      (_.isObject(iamRole) &&
+        ['Fn::Sub', 'Fn::ImportValue', 'Fn::GetAtt'].some((key) => key in iamRole))
```

Comment:
We could also simplift and return true if any key contains `::`, that way it'll be safe against any adjustments in schema

Reply:
good call - changed